pkg/codegen/nodejs/gen.go

```diff
@@ -962,24 +962,26 @@ func (mod *modContext) gen(fs fs) error {
 		fs.add(p, []byte(contents))
 	}
 
-	// Ensure that the target module directory contains a README.md file.
-	readme := mod.pkg.Language["nodejs"].(NodePackageInfo).Readme
-	if readme == "" {
-		readme = mod.pkg.Description
-		if readme != "" && readme[len(readme)-1] != '\n' {
-			readme += "\n"
-		}
-		if mod.pkg.Attribution != "" {
-			if len(readme) != 0 {
+	// Ensure that the top-level (provider) module directory contains a README.md file.
+	if mod.mod == "" {
+		readme := mod.pkg.Language["nodejs"].(NodePackageInfo).Readme
```

Comment:
Nit: might as well move this whole block into the `case "":` in the `switch` statement below.

Reply:
Added this change to my working set, so it'll be in a later PR.